nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -246,6 +272,32 @@ void BtActionServer<ActionT>::executeCallback()
   }
 }
 
+template<class ActionT>
+void BtActionServer<ActionT>::populateErrorCode(
+  typename std::shared_ptr<typename ActionT::Result> result)
+{
+  int highest_priority_error_code = std::numeric_limits<int>::max();
+  for (const auto & error_code_id_name : error_code_ids_) {
+    int current_error_code = std::numeric_limits<int>::max();
+    try {
+      current_error_code = blackboard_->get<int>(error_code_id_name);
```

Comment:
You can do `blackboard_->get<int>("name", var_to_store)` and it return a bool, so you don't need to catch and you don't need to initialize the var_to_store as numeric limits since you only need to check `if (blackboard->...)` to enter the section of code to check the priority level

Reply:
Exceptions are expensive